graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,39 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly."""
+    subselection_inside_fold_scope = inside_fold_scope
+    for directive in ast.directives:
+        name = directive.name.value
+        if name == FilterDirective.name:
+            pass
+        elif name == TagDirective.name:
+            pass
+        elif name == OutputDirective.name:
+            raise GraphQLInvalidMacroError(u'Macros should not have outputs.')
```

Comment:
Let's give some more context in these error messages. Also, `should` -> `must` because it's a requirement.